index.js

```diff
@@ -785,26 +785,56 @@ function buildSingleTypeSerializer (location, input) {
 
 function buildConstSerializer (location, input) {
   const schema = location.schema
-  const type = schema.type
-
-  const hasNullType = Array.isArray(type) && type.includes('null')
-
+  let schemaRef = location.getSchemaRef()
+  if (schemaRef.startsWith(rootSchemaId)) {
+    schemaRef = schemaRef.replace(rootSchemaId, '')
+  }
   let code = ''
 
-  if (hasNullType) {
-    code += `
-      if (${input} === null) {
-        json += 'null'
+  switch (typeof schema.const) {
+    case 'bigint':
+      code += `
+      if (${input} === ${schema.const}n) {
+        json += '${Number(schema.const)}'
       } else {
-    `
-  }
-
-  code += `json += '${JSON.stringify(schema.const)}'`
-
-  if (hasNullType) {
-    code += `
+        throw new Error(\`The value of '${schemaRef}' does not match schema definition.\`)
+      }`
+      break
+    case 'number':
+    case 'boolean':
+      code += `
+      if (${input} === ${schema.const}) {
+        json += ${JSON.stringify(JSON.stringify(schema.const))}
+      } else {
+        throw new Error(\`The value of '${schemaRef}' does not match schema definition.\`)
+      }`
+      break
+    case 'object':
+      if (schema.const === null) {
+        code += `
+        if (${input} === null) {
+          json += 'null'
+        } else {
+          throw new Error(\`The value of '${schemaRef}' does not match schema definition.\`)
+        }`
+      } else {
+        code += `
+        if (JSON.stringify(${input}) === '${JSON.stringify(schema.const)}') {
```

Comment:
This will make a performance downgrade.
But I understand that is a general question: how much the fjs should be a validator? I don't this way. I would use consts only in cases where it can increase performance and don't throw an error.
Example:
```js
if (input === 'constString') {
  return 'constString'
} else {
  return sanitize(input) // heavy computations
}
```

Reply:
Yeah actually that object comparison is hacky. I would instead use a deep compare method. Maybe @lukeed has something.

Reply:
I don't think this is viable

Reply:
I have https://github.com/lukeed/dequal#dequallite if you’re looking to compare two JSON objects, 300b and fast

Reply:
replaced with a method which generates a function to compare the input based on the const value